core/src/main/java/org/apache/iceberg/GenericDataFile.java

```diff
@@ -439,4 +442,29 @@ public String toString() {
     }
     return null;
   }
+
+  private void writeObject(ObjectOutputStream output) throws IOException {
```

Comment:
This looks correct to me, but I'm wondering if we should serialize to a `byte[]` in Avro's binary format. That would cut down on the custom serialization code. Probably not worth doing right now, but we should keep it in mind for later.